feedr-starter-code/js/app.js

```diff
@@ -1,3 +1,351 @@
 /*
   Please add all Javascript code to this file.
-*/
\ No newline at end of file
+*/
+
+// Create 3 functions for each seperate API/AJAX call
+
+    var indexArray = [];
+
+$(document).ready(function(){
+  getDiggData();
+//     $.get( 'http://digg.com/api/news/popular.json', function( r ) {
+//      // We get the data back from the request in the parameter we pass in the function
+//      console.log(r);
+//  });
+    
+    $("#list1").click(function(event){
+          event.preventDefault();
+    console.log("list 1 clicked");
+        $('#main').empty()
+        getDiggData();
+        
+    });
+    $("#list2").click(function(event){
+          event.preventDefault();
+    console.log("list 2 clicked");
+        $('#main').empty()
+        getNewsApiData();
+        
+    });
+      $("#list3").click(function(event){
+          event.preventDefault();
+    console.log("list 3 clicked");
+        $('#main').empty()
+        getMashableData();
+        });
+    
+    
+    
+//     $('.article').on('click', function() { 
+//         
+//                     console.log("article clicked!!!!!!!");
+//         console.log(this)
+//         $("#popUp").removeClass( "loader hidden" )
+//         //var retval = []
+//         $('.article').each(function(){
+//            // console.log(this)
+//             retval.push($(this).find('featuredImage').attr('section'))
+//         })
+//        // return console.log(retval);
+//         //console.log(retval);
+//        
+////        $("#popUp").style.display = 'flex';
+//        
+////         var popup = document.getElementById("#popUp");
+////  popup.classList.toggle(".show");
+//
+// });
+
+    $('.closePopUp').click(function(event){
+                    console.log("x clicked");
+        $("#popUp").addClass( "loader hidden" );
+
+    })
+    
+//    $("#list1").on('click', function(event) {
+//  event.preventDefault();
+//    console.log("list 1 clicked");
+//
+//});
+    
+});
+
+function getDiggData(){
+           console.log("in");
+
+     $.get('https://accesscontrolalloworiginall.herokuapp.com/http://digg.com/api/news/popular.json')
+     .then(response => {
+         var feedArray = response.data.feed;
+         fillArticles(feedArray);
+             //   console.log(response.data.feed);
+
+//         feedArray.forEach(function(element) {
+//            console.log(element.content.title);
+//            console.log(element.content.domain_name);
+//            console.log(element.content.description);
+//            console.log(element.content.media.images[3].url);
+
+         
+     });
+}
+    console.log(indexArray.length);
+
+//news api url need to be update every day
+ function getNewsApiData(){
+             console.log("in newsApi");
+
+     $.get('https://accesscontrolalloworiginall.herokuapp.com/https://newsapi.org/v2/everything?q=bitcoin&sortBy=publishedAt&apiKey=e75626217a1249919184dd5d7679e3e7')
```

Comment:
It's better if you put the API key in an external file and not push it to Github!